rust-crates/symblib/src/gosym/mod.rs

```diff
@@ -540,3 +540,55 @@ fn range_rel2abs(base: VirtAddr, rng: Range<raw::TextStartOffset>) -> Range<Virt
         end: base.wrapping_add(rng.end.0),
     }
 }
+
+impl symbconv::PointResolver for GoRuntimeInfo<'_> {
+    fn symbols_for_pc(&self, pc: VirtAddr) -> symbconv::Result<Vec<symbconv::ResolvedSymbol>> {
+        let mut symbols = Vec::new();
+
+        match self.find_func(pc) {
+            Ok(Some(func)) => {
+                let mut source_files = Vec::new();
+                let mut line_numbers = Vec::new();
+
+                // For file mappings
+                let mut file_iter = func
+                    .file_mapping()
+                    .map_err(|e| symbconv::Error::Go(symbconv::go::Error::Gosym(e)))?;
+                while let Ok(Some((range, file))) = file_iter.next() {
+                    if range.contains(&VirtAddr::from(pc)) {
+                        source_files.push(file.unwrap_or("<unknown>").into());
```

Comment:
I think you can break here after finding the first one, and turn `source_files` into `Option<String>` instead. I don't think that a symbol can have more than one filename?

Reply:
Don't remember exactly why I went with `Vec<_>` instead of `Option<_>` for files and line numbers. Dropped this approach and followed the suggestion with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/380/commits/a73f6f80777d80375c5e90d314ef484751185569.